include/dlaf/factorization/qr/api.h

```diff
@@ -10,9 +10,12 @@
 
 #pragma once
 
+#include <vector>
+
 #include <pika/execution.hpp>
 
 #include <dlaf/communication/communicator_pipeline.h>
+#include <dlaf/factorization/qr/internal/get_tfactor_nworkers.h>
```

Comment:
Needed here?

Reply:
Good that you raised it 😄 This is a choice that I wanted to quickly discuss.

Workspaces tiles have to be allocated outside, and there should be enough tiles for potentially all the workers. So, we are sure that the user needs to check how many workers there are through this getter, so I opted for including it together with the declarations of the API where the workspace is going to be passed.

Reply:
I think internally we can explicitly include the required header? I think it'd be a different discussion if we considered a "public API header"-to-"names" mapping, but for internal functionality I think directly including the header that actually provides a function is simplest. I'm not insisting on this at all though, that's just my subjective opinion.

Reply:
it makes totally sense! see https://github.com/eth-cscs/DLA-Future/pull/1214/commits/80797c1842d7d822b971e60868d51d694e2305be